recipe_scrapers/_utils.py

```diff
@@ -202,17 +221,15 @@ def url_path_to_dict(path):
         r"^"
         r"((?P<schema>.+?)://)?"
         r"((?P<user>.+?)(:(?P<password>.*?))?@)?"
-        r"(?P<host>.*?)"
-        r"(:(?P<port>\d+?))?"
+        r"(?P<host>[^:/]+)"
+        r"(:(?P<port>\d+))?"
         r"(?P<path>/.*?)?"
-        r"(?P<query>[?].*?)?"
+        r"(?P<query>\?.*?)?"
```

Comment:
```suggestion
        r"(?P<query>[?].*?)?"
```

I prefer the existing pattern syntax for this line, if that's OK with you.